codalab/lib/metadata_util.py

```diff
@@ -60,16 +60,19 @@ def add_edit_argument(parser):
     )
 
 
-def request_missing_metadata(bundle_subclass, args, initial_metadata=None):
+def get_default_initial_metadata(bundle_subclass, args):
+    return {
+        spec.key: getattr(args, metadata_key_to_argument(spec.key,))
+        for spec in bundle_subclass.get_user_defined_metadata()
+    }
+
+
+def fill_missing_metadata(bundle_subclass, args, initial_metadata=None):
     '''
-    For any metadata arguments that were not supplied through the command line,
-    pop up an editor and request that data from the user.
+    Fill in the missing metadata with default values and return modified metadata
     '''
     if not initial_metadata:
-        initial_metadata = {
-          spec.key: getattr(args, metadata_key_to_argument(spec.key,))
-          for spec in bundle_subclass.get_user_defined_metadata()
-        }
+        initial_metadata = get_default_initial_metadata(bundle_subclass, args)
```

Comment:
Move this logic to get_missing_metadata in bundle_cli, so that initial_metadata is never None